### PR TITLE
2023q3: dpaa2: Status update

### PR DESCRIPTION
**Pull request by dsalychev**

### CODE REVIEW

File: website/content/en/status/report-2023-07-2023-09/dpaa2.adoc [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+link:https://cgit.freebsd.org/src/tree/sys/dev/dpaa2[DPAA2 in the FreeBSD source tree] +
+link:https://github.com/mcusim/freebsd-src[DPAA2 on Github]
+
+Contact: Dmitry Salychev <dsl@FreeBSD.org> +
+Contact: Bjoern A. Zeeb <bz@FreeBSD.org>
+
+==== What is DPAA2?
+
+DPAA2 is a hardware-level networking architecture found in some NXP SoCs which contain hardware blocks including Management Complex (MC, a command interface to manipulate DPAA2 objects), Wire Rate I/O processor (WRIOP, packets distribution, queuing, drop decisions),
+Queues and Buffers Manager (QBMan, Rx/Tx queues control, Rx buffer pools) and the others. The Management Complex runs NXP-supplied firmware which provides DPAA2 objects as an abstraction layer over those blocks to simplify an access to the underlying hardware.
```
Comment (grahamperrin) [inline website/content/en/status/report-2023-07-2023-09/dpaa2.adoc:13]:
```suggestion
DPAA2 is a hardware-level networking architecture found in some NXP SoCs which contain hardware blocks including Management Complex (MC, a command interface to manipulate DPAA2 objects), Wire Rate I/O processor (WRIOP, packets distribution, queuing, drop decisions), Queues and Buffers Manager (QBMan, Rx/Tx queues control, Rx buffer pools) and others.
The Management Complex runs NXP-supplied firmware which provides DPAA2 objects as an abstraction layer over those blocks to simplify access to the underlying hardware.
```

File: website/content/en/status/report-2023-07-2023-09/dpaa2.adoc [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+* Panic under heavy network load link:https://github.com/mcusim/freebsd-src/issues/19[fixed].
+
+* FDT/ACPI MDIO support.
+
+* NFS root mount link:https://github.com/mcusim/freebsd-src/issues/7[doesn't hang] on netboot over DPAA2 anymore.
+
+* Drivers link:https://github.com/mcusim/freebsd-src/issues/2[started] to communicate with MC via their own command portals (DPMCP).
+
+* link:https://github.com/mcusim/freebsd-src/issues?q=is%3Aissue+is%3Aclosed[List of all closed issues].
```
Comment (grahamperrin) [inline website/content/en/status/report-2023-07-2023-09/dpaa2.adoc:27]:
```suggestion
* Isolation between DPAA2 channels is link:https://cgit.freebsd.org/src/commit/?id=58983e4b0253ad38a3e1ef2166fedd3133fdb552[improved].
* Panic under heavy network load is link:https://github.com/mcusim/freebsd-src/issues/19[fixed].
* FDT/ACPI MDIO support.
* NFS root mount link:https://github.com/mcusim/freebsd-src/issues/7[no longer hangs] on netboot over DPAA2.
* Drivers link:https://github.com/mcusim/freebsd-src/issues/2[started] to communicate with MC via their own command portals (DPMCP).
* link:https://github.com/mcusim/freebsd-src/issues?q=is%3Aissue+is%3Aclosed[List of all closed issues].
```

File: website/content/en/status/report-2023-07-2023-09/dpaa2.adoc [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+==== Work in Progress
+
+* Work on link:https://cgit.freebsd.org/src/commit/?id=2a9021898c4ee2154787da862c238cfeccd655df[dev/sff] started to support SFF/SFP modules in order to test DPAA2 drivers on the links above 1 Gbit/s
```
Comment (grahamperrin) [inline website/content/en/status/report-2023-07-2023-09/dpaa2.adoc:31]:
```suggestion
Work on link:https://cgit.freebsd.org/src/commit/?id=2a9021898c4ee2154787da862c238cfeccd655df[dev/sff] started to support SFF/SFP modules in order to test DPAA2 drivers on links above 1 Gbit/s.
```

File: website/content/en/status/report-2023-07-2023-09/dpaa2.adoc [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+==== Plan
+
+* Heavy network load tests (2.5 Gbit/s, 10 Gbit/s) and bottlenecks mitigation.
+
+* Cached memory-backed software portals.
+
+* Driver resources de-allocation to unload dpaa2.ko properly.
+
+* Further parts (DPSW, DCE, etc.) supported by the hardware.
```
Comment (grahamperrin) [inline website/content/en/status/report-2023-07-2023-09/dpaa2.adoc:41]:
```suggestion
* Heavy network load tests (2.5 Gbit/s, 10 Gbit/s) and bottlenecks mitigation.
* Cached memory-backed software portals.
* Driver resources de-allocation to unload `dpaa2.ko` properly.
* Further parts (DPSW, DCE, etc.) supported by the hardware.
```

**Review by grahamperrin — CHANGES_REQUESTED** [review]

File: website/content/en/status/report-2023-07-2023-09/dpaa2.adoc (hunk 1 above)
Comment (grahamperrin) [reply website/content/en/status/report-2023-07-2023-09/dpaa2.adoc:13]: > … blocks including … Management Complex …, Queues and Buffers Manager (…) and others.

That's _including_, not _comprising_, so you can lose the two words at the tail of the sentence: 

> … blocks including … Management Complex …, and Queues and Buffers Manager (…).

**Comment by dsalychev** [comment]
Uhh, I've force-pushed changes... sorry. @grahamperrin Could you take a look?

File: website/content/en/status/report-2023-07-2023-09/dpaa2.adoc [hunk 5]
```diff
@@ -0,0 +1,36 @@
+=== NXP DPAA2 support
+
+Links +
+link:https://cgit.freebsd.org/src/tree/sys/dev/dpaa2[DPAA2 in the FreeBSD source tree] +
+link:https://github.com/mcusim/freebsd-src[DPAA2 on Github]
```
Comment (grahamperrin) [inline website/content/en/status/report-2023-07-2023-09/dpaa2.adoc:5]:
```suggestion
Links: +
link:https://cgit.freebsd.org/src/tree/sys/dev/dpaa2[DPAA2 in the FreeBSD source tree] URL: link:https://cgit.freebsd.org/src/tree/sys/dev/dpaa2[] +
link:https://github.com/mcusim/freebsd-src[DPAA2 on Github] URL: link:https://cgit.freebsd.org/src/tree/sys/dev/dpaa2[]
```

Whilst the repetitions, prefixes etc. do appear peculiar, some peculiarities are currently required for the eventual plain text edition of the report to include visible or usable links. See, for example, last quarter's:

* https://lists.freebsd.org/archives/freebsd-announce/2023-July/000077.html

**Review by grahamperrin — CHANGES_REQUESTED** [review]

**Comment by grahamperrin** [comment]
> Uhh, I've force-pushed changes... sorry. 

IMHO (drive-by reviewer, not a status team member): no harm done, but thanks.

File: website/content/en/status/report-2023-07-2023-09/dpaa2.adoc [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+Work on link:https://cgit.freebsd.org/src/commit/?id=2a9021898c4ee2154787da862c238cfeccd655df[dev/sff] started to support SFF/SFP modules in order to test DPAA2 drivers on links above 1 Gbit/s.
+
+==== Plan
+
+* Heavy network load tests (2.5 Gbit/s, 10 Gbit/s) and bottlenecks mitigation.
+* Cached memory-backed software portals.
+* Driver resources de-allocation to unload dpaa2.ko properly.
+* Further parts (DPSW, DCE, etc.) supported by the hardware.
+
+Sponsor: Bare Enthusiasm :) +
```
Comment (lsalvadore) [inline website/content/en/status/report-2023-07-2023-09/dpaa2.adoc:37]: I am sure that this is one of the most important sponsors, but I discourage listing it: I guess it holds for almost any project, it does not give any new real information, and there is also the risk that the importance of the real economical sponsor is underestimated.

**Review by lsalvadore — CHANGES_REQUESTED** [review]
A few changes and it is ready.

File: website/content/en/status/report-2023-07-2023-09/dpaa2.adoc [hunk 7]
```diff
@@ -0,0 +1,37 @@
+=== NXP DPAA2 support
+
+Links: +
+link:https://cgit.freebsd.org/src/tree/sys/dev/dpaa2[DPAA2 in the FreeBSD source tree] +
+URL: link:https://cgit.freebsd.org/src/tree/sys/dev/dpaa2[] +
+link:https://github.com/mcusim/freebsd-src[DPAA2 on Github] +
+URL: link:https://github.com/mcusim/freebsd-src[]
```
Comment (lsalvadore) [inline website/content/en/status/report-2023-07-2023-09/dpaa2.adoc:7]:
```suggestion
link:https://cgit.freebsd.org/src/tree/sys/dev/dpaa2[DPAA2 in the FreeBSD source tree] URL: link:https://cgit.freebsd.org/src/tree/sys/dev/dpaa2[] +
link:https://github.com/mcusim/freebsd-src[DPAA2 on Github] URL: link:https://github.com/mcusim/freebsd-src[]
```

**Review by lsalvadore — APPROVED** [review]
I have found one last small thing that you can fix at commit time.
Approved, thanks.

**Comment by grahamperrin** [comment]
https://github.com/freebsd/freebsd-doc/pull/259#issuecomment-1732544892

> Uhh, I've force-pushed changes... sorry. 

<https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=273895> intended to clarify the situation, and is closed as fixed, however I don't see the fix. 

@dsalychev sorry.

**Comment by lsalvadore** [comment]
Merged in commit 7a4522d6295fddc50799a4e93864d0c8baf22f6e, thanks.

**Pull request closed without merging**